### DISCUSSION TITLE
firefox: How to allow /dev/hidraw0?

### DISCUSSION THREAD

**Discussion by pjhfggij** [netblue30/firejail#6530] [Q&A] [answered]
I've tried adding /dev/hidraw0 to firefox.local profile `whitelist /dev/hidraw0`, but this removed all other devices in /dev.

Is there a way to add a single device without removing the rest?

**Comment by rusty-snake** [comment]
> Is there a way to add a single device without removing the rest?

No.

> how to add /dev/hidraw0 to firefox

The recommended way to allow U2F devices is to set `browser-disable-u2f no` in `/etc/firejail/firejail.config`.

https://github.com/netblue30/firejail/blob/4ded6b77741894df3699946f3a9bb113e0ca6c84/etc/firejail.config#L25-L26

**Reply by pjhfggij** [reply]
thanks for the quick reply, I haven't changed this setting, so not sure why it just stopped working. It still works fine when run without firejail. Upon request to touch the device nothing happens, request doesn't seem to reach the device. The only changes in firefox.local (librewolf):

`firejail --nowhitelist=$HOME/.librewolf librewolf`

```
# for browserpass extension
whitelist ${HOME}/.password-store
noblacklist ${HOME}/.password-store
whitelist ${HOME}/.gnupg
noblacklist ${HOME}/.gnupg
writable-run-user
whitelist ${RUNUSER}/gnupg

nowhitelist ${HOME}/.kde/share/config/kdeglobals
tmpfs ${HOME}/.kde
tmpfs ${HOME}/.cache
nowhitelist ${HOME}/.themes
tmpfs ${HOME}/.themes
nowhitelist ${HOME}/.pki
nowhitelist ${HOME}/.fonts.conf
nowhitelist ${HOME}/.fonts
nowhitelist ${HOME}/.gnome2
nowhitelist ${HOME}/.gtkrc-2.0
tmpfs ${HOME}/.config
tmpfs ${HOME}/.local
blacklist ${HOME}/.Xauthority
```

however /etc/firejail/librewolf.profile and /usr/bin/firejail hasn't changed since May this year, so perhaps this isn't a firejail issue?

**Comment by kmk3** [comment, marked as answer]
> I've tried adding /dev/hidraw0 to firefox.local profile `whitelist
> /dev/hidraw0`, but this removed all other devices in /dev.
> 
> Is there a way to add a single device without removing the rest?

Does it show up with the following in firefox.local?

```
ignore nou2f
```

If not, does it work with:

```
ignore nou2f
ignore private-dev
```

If not, does it work with the above when using firejail-git?

**Reply by pjhfggij** [reply]
thank you @kmk3, you are right,  `ignore nou2f` made hidraw (used by nitrokey) accessible to firefox (used by librewolf) again
I don't know why it suddenly stopped working, I don't recall changing any settings, glad it's working again though, this saves me a few keystrokes when using web browser under firejail each time
